clippy_lints/src/non_octal_unix_permissions.rs

```diff
@@ -39,6 +41,24 @@ declare_clippy_lint! {
 
 declare_lint_pass!(NonOctalUnixPermissions => [NON_OCTAL_UNIX_PERMISSIONS]);
 
+fn check_binary_unix_permissions(lit_kind: &LitKind, snip: &str) -> bool {
+    // support binary unix permissions
+    if let Some(num_lit) = NumericLiteral::from_lit_kind(snip, lit_kind) {
+        if num_lit.radix != Radix::Binary {
+            return false;
+        }
+
+        let group_sizes: Vec<usize> = num_lit.integer.split('_').map(str::len).collect();
+        // check whether is binary format unix permissions
+        if group_sizes.len() != 3 {
```

Comment:
We should allow lengths of `4` as well, for the setuid, setgid and sticky bits.

Reply:
Oh yeah, I forgot that. Fixed.